draft-jholland-quic-multicast.md

```diff
@@ -172,6 +173,9 @@ The server ensures that in aggregate, all channels that the client has currently
 The client sends back information about how it has responded to the server's requests to join and leave channels in MC_CLIENT_CHANNEL_STATE ({{client-channel-state-frame}}) frames.
 MC_CLIENT_CHANNEL_STATE frames are only sent for channels after the server has requested the client to join the channel, and are thereafter sent any time the state changes.
 
+Clients MUST make sure that they allow enough time for the initial creation of the multicast forwarding tree before determining if reception of data on the multicast channel is possible or not.
```

Comment:
I think this is a SHOULD, I think the timing will be too fuzzy for a MUST here.

We might need a section or a reference laying out the issues better.  I think 2xRTT is probably too short, RTT to the server is likely to be much faster than the initial join propagation, since that has hop-by-hop state changes and probably tunnel establishment.  I suspect the right answer is to have an administratively configured default for the client and recommend that it adopt a network-specific recommended initial join delay if one is available, with a default of something like 15s if the local receive network does not provide a recommendation.  The initial join can be quite slow sometimes, especially if there is packet loss somewhere.

Reply:
Another option here is to leave it up to the server by default, and say the client MAY leave with a new reason according to a separate timer it that can be configured either administratively or according to advice from the local receive network, but describe that it should be long enough to allow for lost packets during hop-by-hop state establishment for the network path to the multicast sender, which can be slow.

I kind of like this idea better, I don't think client ever has to leave if no traffic comes, but it MAY have its own override if it wants, and server can of course decide to give up too.